server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -492,6 +519,14 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
                 } else {
                     return point.parseGeoHash(geohash, effectivePoint);
                 }
+            } else if (coordinates != null) {
+                if (geojsonType == null || geojsonType.toLowerCase(Locale.ROOT).equals("point") == false) {
```

Comment:
We need to check that we have no other elements in GeoJson (lat, lon or geohash)

Reply:
Agreed!